uwblueprint/headsupguys
# [Tool] Build tool page

## Notion ticket link

[[Tool] Build tool page](https://www.notion.so/uwblueprintexecs/Tool-Build-tool-page-259badfdb1734afa93cfca3bba993ecf)



## Implementation description
* Built tool page
* Added util helper function for `useMediaQuery` with fixed screen size breakpoint
* Used `ReactPlayer` for displaying video
* Split styling for buttons into `AdminButton` and `Button` component themes
* Added tool layout with navbar



## Steps to test
1. From `http://localhost:3000/` click on 'Continue Learning' for any tool (should take you to `http://localhost:3000/tool?id=[id]`)
2. Observe tool page (please compare to [Figma](https://www.figma.com/file/GhDd9C0XsyVDgwL5zzvDvO/HUG-W22?node-id=4355%3A55348)), should look something like this:

<img width="600" alt="Screen Shot 2022-02-24 at 8 36 01 PM" src="https://user-images.githubusercontent.com/49915445/155636956-33fd0c81-1873-4401-b505-524b0238c066.png"> <img width="160" alt="Screen Shot 2022-02-24 at 8 36 24 PM" src="https://user-images.githubusercontent.com/49915445/155636962-d6bcb6d2-2df2-44ba-a7b6-6b3876458e8c.png">

4. Scroll on page to check that navbar and footer are sticky
5. Observe that the video player is present (the video link may or may not be broken)
6. Observe that there is a description below the video
7. Expand and collapse all additional resources tabs and make sure that show the resource links
8. Click on some resource links and make sure they open in a new tab
9. Observe that there are recommended tool cards
10. Click on 'Start Module' and observe redirect to `http://localhost:3000/module?id=[id]` (note the id in the url is unique, and the link does not yet work)
11. Click on 'Take Self-Check' and observe redirect to `http://localhost:3000/selfCheck?id=[id]` (note the id in the url is unique, and the link does not yet work)
12. Close footer and check that it disappears
13. Click on the certificate icon in the header and check that it toggles a banner
14. Click on 'here' in the banner text and observe redirect to `http://localhost:3000/certificate?id=[id]`
15. Check that the 'Your Module Progress' section has a 100% full progress bar
16. Click the restart icon and check that a modal is opened
17. Click on 'No, cancel' or 'Yes, restart' to close the modal (note that restarting does nothing right now)
18. Click on 'Review Module' and observe redirect to `http://localhost:3000/module?id=[id]`(note the id in the url is unique, and the link does not yet work)
19. Click on 'Take Self-Check' and observe redirect to `http://localhost:3000/selfCheck?id=[id]`(note the id in the url is unique, and the link does not yet work)
20. In the code, search for `progress = 100` in the file `pages/tool.tsx`, and change it to `progress = 21`
21. Check that the 'Your Module Progress' section has a 21% full progress bar and the green button says 'Continue Module'
22. Click on the certificate icon in the header and check that it toggles a banner, and the banner no longer has a 'here' link
23. In the code, search for `progress = 21` in the file `pages/tool.tsx`, and change it to `progress = 0`
24. Check that the 'Your Module Progress' section is now titled 'Interested in learning more about ...', there is no progress bar, and the green button says 'Start Module' (note that this section should be relocated below the video and description for mobile layouts)
25. Repeat steps 2-24 with a mobile layout (resize your window horizontally)
26. Also test more than one module, and also a newly created module



## What should reviewers focus on?
* Does it work?
* Does the styling look ok?
* Does the code look ok?


## Checklist
